monai/transforms/spatial/array.py

```diff
@@ -2609,3 +2620,167 @@ def _get_params(
+        # create the patch iterator which sweeps the image row-by-row
+        array_np, *_ = convert_data_type(array, np.ndarray)
+        patch_iterator = iter_patch(
+            array_np,
+            patch_size=(None,) + self.patch_size,  # expand to have the channel dim
+            start_pos=(0,) + self.start_pos,  # expand to have the channel dim
+            overlap=self.overlap,
+            copy_back=False,
+            mode=self.pad_mode,
+            **self.pad_opts,
+        )
+        if self.sort_fn is not None:
+            output = sorted(patch_iterator, key=self.sort_fn)
+        else:
+            output = list(patch_iterator)
+        if self.fix_num_patches:
+            output = output[: self.fix_num_patches]
+            if len(output) < self.fix_num_patches:
+                if self.pad_opts.get("constant_values"):
+                    patch = np.ones((array.shape[0], *self.patch_size)) * self.pad_opts["constant_values"]
+                else:
+                    patch = np.zeros((array.shape[0], *self.patch_size))
+                location = np.zeros((3, len(self.patch_size)))
+                output += [(patch, location)] * (self.fix_num_patches - len(output))
+
+        output = [convert_to_dst_type(src=patch, dst=array)[0] for patch in output]
+
+        return output
+
+
+class RandGridPatch(GridPatch, RandomizableTransform):
+    """
+    Extract all the patches sweeping the entire image in a row-major sliding-window manner with possible overlaps,
+    and with random offset for the starting position of upper left corner of the image.
+    It can sort the patches and return all or a subset of them.
+
+    Args:
+        patch_size: size of patches to generate slices for, 0 or None selects whole dimension
+        min_start_pos: the minimum range of starting position to be selected randomly. Defaults to 0.
+        max_start_pos: the maximum range of starting position to be selected randomly.
+            Defaults to image size modulo patch size.
+        fix_num_patches: number of patches to return. Defaults to None, which returns all the available patches.
+        overlap: the amount of overlap of neighboring patches in each dimension (a value between 0.0 and 1.0).
+            If only one float number is given, it will be applied to all dimensions. Defaults to 0.0.
+        sort_fn: a callable or string that defines the order of the patches to be returned. If it is a callable, it
+            will be passed directly to the `key` argument of `sorted` function. The string can be "min" or "max",
+            which are, respectively, the minimum and maximum of the sum of intensities of a patch across all dimensions
+            and channels. Also "random" creates a random order of patches.
+            By default no sorting is being done and patches are returned in a row-major order.
+        pad_mode: {``"constant"``, ``"edge"``, ``"linear_ramp"``, ``"maximum"``, ``"mean"``,
```

Comment:
You could refer to NumpyPadMode and PytorchPadMode instead of listing the values here to save a little space if you wanted.